File: API/src/main/java/br/cefet/pechinchatech/model/ProdutoBase.java
package br.cefet.pechinchatech.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProdutoBase {
    @NotNull(message = "Insira um score!")
    int score_userbenchmark;

    @NotNull(message = "Insira um rank!")
    int rank_userbenchmark;

    Long id;

    @NotBlank(message = "Insira um nome!")
    String nome;

    @NotNull(message = "Insira um tipo!")
    TipoProduto tipo;

    @NotNull(message = "Insira um número da linha csv!")
    int csv_row;

    @NotBlank(message = "Insira uma marca!")
    @Size(max = 16)
    String marca;
}

File: Site/PechinchaTech/src/app/model/produto-base.ts
enum Tipo {
    "SSD",
    "HDD",
    "GPU",
    "CPU",
    "RAM"
}

export class ProdutoBase {
    id: number;
    nome: string;
    csv_row: number;
    tipo: Tipo;
    marca: string;
    rank_userbenchmark: number;
    score_userbenchmark: number;

    constructor() {
        this.id = 0;
        this.nome = "";
        this.csv_row = 0;
        this.tipo = Tipo.SSD;
        this.marca = "";
        this.rank_userbenchmark = 0;
        this.score_userbenchmark = 0;
    }
}
